File: src/main/java/de/tectoast/emolga/commands/pokemon/NaturalGiftCommand.java
package de.tectoast.emolga.commands.pokemon;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.records.NGData;
import de.tectoast.emolga.utils.sql.DBManagers;
import net.dv8tion.jda.api.EmbedBuilder;

import java.awt.*;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

public class NaturalGiftCommand extends Command {

    public NaturalGiftCommand() {
        super("naturalgift", "Entweder alle Beeren für einen Typ, oder den Typen von einer Beere", CommandCategory.Pokemon);
        setArgumentTemplate(ArgumentManagerTemplate.builder()
                .add("stuff", "Sache", "Entweder ein Typ oder ein Beerenname", Translation.Type.of(Translation.Type.ITEM, Translation.Type.TYPE))
                .setExample("!naturalgift Water")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) {
        Translation t = e.getArguments().getTranslation("stuff");
        String translation = t.getTranslation();
        if (t.isFromType(Translation.Type.ITEM)) {
            NGData ngData = DBManagers.NATURAL_GIFT.fromName(translation);
            e.reply(new EmbedBuilder()
                    .setTitle(translation)
                    .addField("Typ", ngData.type(), false)
                    .addField("Basepower", String.valueOf(ngData.bp()), false)
                    .setColor(Color.CYAN)
                    .build()
            );
        } else {
            List<NGData> ngData = DBManagers.NATURAL_GIFT.fromType(translation);
            e.reply(new EmbedBuilder()
                    .setTitle(translation)
                    .setDescription(
                            ngData.stream().sorted(Comparator.comparing(NGData::bp)).map(d -> d.name() + "/" + getEnglName(d.name()) + ": " + d.bp()).collect(Collectors.joining("\n"))
                    )
                    .setColor(Color.CYAN)
                    .build()
            );
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/pokemon/NaturalGiftCommand.kt
package de.tectoast.emolga.commands.pokemon

import de.tectoast.emolga.commands.*
import de.tectoast.emolga.database.exposed.NaturalGift
import dev.minn.jda.ktx.messages.Embed

class NaturalGiftCommand : Command(
    "naturalgift", "Entweder alle Beeren für einen Typ, oder den Typen von einer Beere", CommandCategory.Pokemon
) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder().add(
            "stuff",
            "Sache",
            "Entweder ein Typ oder ein Beerenname",
            Translation.Type.of(Translation.Type.ITEM, Translation.Type.TYPE)
        ).setExample("!naturalgift Water").build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val t = e.arguments.getTranslation("stuff")
        val translation = t.translation
        if (t.isFromType(Translation.Type.ITEM)) {
            val ngData =
                NaturalGift.byName(translation) ?: return e.reply("Dieses Item hat keinen Effekt auf Natural Gift!")
            e.reply(
                Embed {
                    title = translation
                    field {
                        name = "Typ"
                        value = ngData.type
                        inline = false
                    }
                    field {
                        name = "Basepower"
                        value = ngData.bp.toString()
                        inline = false
                    }
                    color = embedColor
                })
        } else {
            e.reply(Embed(title = translation,
                color = embedColor,
                description = NaturalGift.byType(translation)
                    .sortedBy { it.bp }.joinToString("\n") {
                        "${it.name}/${getEnglName(it.name)}: ${it.bp}"
                    })
            )
        }
    }
}
